src/rxjs_wrapper.ts

```diff
@@ -70,61 +65,99 @@ const unpack = <T>(event: T | Box<T>, observable: string): { id: number, event:
+    return (source$: Observable<T>) => {
+        console.log(`wrapPipeableOperator ${metadata.line} ${metadata.function} ${metadata.observable}`);
+        const message = createPayloadMessage(metadata, MessageType.oprator);
+        sendToBackpage(message);
+
+        let id: number;
+        return source$.pipe(
+            map(e => unpack(e, metadata.observable)),
+            map(e => {
+                id = e.id;
+                return e.event;
+            }),
+            operatorFn,
+            tap(e => sendToBackpage(createPayloadMessage<T>(createEvent(e, metadata.observable, id), MessageType.event))),
+            tap(e => console.log(`${e} ${metadata.observable} ${id}`)),
+            map(e => last ? e : new Box<T>(e, id))
+        );
+    };
+};
 
-    let id: number;
-    return source$.pipe(
-        map(e => unpack(e, metadata.observable)),
-        map(e => {
-            id = e.id;
-            return e.event
-        }),
-        operatorFn,
-        tap(e => sendToBackpage(createPayloadMessage<T>(createEvent(e, metadata.observable, id), MessageType.event))),
-        tap(e => console.log(`${e} ${metadata.observable} ${id}`)),
-        map(e => last ? e : new Box<T>(e, id))
-    );
+// Wrap and return pipe statement.
+export const wrapPipe = <T>(source$: Observable<T>, metadata: PipeMetadata, ...operators: []) => {
+    console.log(`wrapped pipe identifier: ${metadata.identifier}  observable: ${metadata.observable} uuid: ${metadata.uuid}`);
+    const message = createPayloadMessage(metadata, MessageType.pipe);
+    sendToBackpage(message);
+    return source$.pipe(...operators);
 };
 
 type Next<T> = (data: T) => void;
 type Error<E> = (error: E) => void;
 type Complete = () => void;
 
+// Create SubscribeEvent payload message.
+const createSubscribeEventMessage = <T>(type: SubscribeEvent<T>['type'], data: T, observable: string) => {
+    const subscribeEvent: SubscribeEvent<T> = { type, data, observable };
+    const message = createPayloadMessage(subscribeEvent, MessageType.subscribeEvent);
+    return message;
+};
+
 // Wrap subscribe and its optional next, error and complete arguments.
-export const wrapSubscribe = <T, E>(source$: Observable<T>, next?: Next<T>, error?: Error<E>, complete?: Complete): Subscription => {
+export const wrapSubscribe = <T, E>(
+    source$: Observable<T>,
+    metadata: SubscriberMetadata,
+    next?: Next<T>,
+    error?: Error<E>,
+    complete?: Complete
+): Subscription => {
+    console.log(`wrapped subscribe ${metadata.observable} pipe ${metadata.pipes[0]}`);
+    const message = createPayloadMessage(metadata, MessageType.subscribe);
+    sendToBackpage(message);
+
     let [wrappedNext, wrappedError, wrappedComplete] = [null, null, null];
 
     if (next) {
         wrappedNext = (event: T) => {
+            const eventMessage = createSubscribeEventMessage('NEXT', event, metadata.observable);
```

Comment:
You have limited string type for the type property on SubscribeEvent, why not use an Enum?